### ISSUE TITLE
Instagram API sync failed repeatedly and now only syncing 10% of records

### ISSUE THREAD

**Issue by jimbeepbeep** [airbytehq/airbyte#4028]
## Expected Behavior
Instagram syncs all  tables to BigQuery as usual

## Current Behavior
My syncs stopped working yesterday and repeatedly failed, so I figured the best approach would be to upgrade my Airbyte and Instagram connector versions.  After failing to successfully do this I eventually fragged the VM and reinstalled from scratch, then re-set up the connection to the same dataset and set up the sync.

Weirldy now it seems to only be syncing ~10k records when it was successfully syncing ~100k records before, only the `media` table has any new data.  Also the logs look like there are loads of empty responses, where prviously you could see the content of the post captions in the logs.

I have checked my access token with Facebook's debugger and it looks fine:

![Screen Shot 2021-06-10 at 11 33 05 am REDACTED](https://user-images.githubusercontent.com/34912260/121537608-5c2c0200-ca04-11eb-8165-6e570f3c766a.png)

### Pre-upgrade (Airbyte 0.19.0-alpha, Instagram 0.1.4)
[logs-144-0.txt](https://github.com/airbytehq/airbyte/files/6631800/logs-144-0.txt)
[logs-144-1.txt](https://github.com/airbytehq/airbyte/files/6631798/logs-144-1.txt)
[logs-144-2.txt](https://github.com/airbytehq/airbyte/files/6631797/logs-144-2.txt)

### Post-upgrade (Airbyte 0.24.8-alpha, Instagram 0.1.5)
[logs-2-0.txt](https://github.com/airbytehq/airbyte/files/6631855/logs-2-0.txt)

## Steps to Reproduce
1. Select all tables to sync 
1. All `Full refresh - Append` (except user_insights as `Incremental - Append`)
1. start_date: 2021-01-01T00:00:00Z

## Severity of the bug for you
High

## Airbyte Version
0.24.8-alpha

## Connector Version (if applicable)
Instagram 0.1.5

## Additional context
Deployed to a GCP VM (Compute Engine) n1-standard-1 (1 vCPU, 3.75 GB memory)

**Comment by jimbeepbeep** [comment]
I have tried to re-build my dataset by creating an entirely new dataset in BigQuery and new Instagram connection and although the Sync indicates that it has Succeeded, only the media table is populated, and only with data from 20 accounts (there are 93 active accounts linked to this access token.  Also all of the _airbyte_raw prefixed tables are empty (except `_airbyte_raw_media`).  Logs for the new Sync are included below:

[logs-16-0.txt](https://github.com/airbytehq/airbyte/files/6647479/logs-16-0.txt)

@yevhenii-ldv / @sherifnada  I seem to recall this is similar to some of the behaviour encountered when initially developing and testing this connector ([#1814](https://github.com/airbytehq/airbyte/issues/1814)), let me know if there's anything I can do to help as I can't seem to get it working at all and it is critical for my client

**Comment by jimbeepbeep** [comment]
Additional context on sync status:

table name | data sync
--- | :-:
_airbyte_raw_media|Y
_airbyte_raw_media_insights |N
_airbyte_raw_stories|N 
_airbyte_raw_story_insights|N
_airbyte_raw_user_insights|N
_airbyte_raw_user_lifetime_insights|N
_airbyte_raw_users|N
media|Y
media_children|Y
media_children_owner|Y
media_insights|N
media_owner|Y
stories|N
stories_owner|N
story_insights|N
user_insights|N
user_lifetime_insights|N
users|N

Also tracking distinct usernames (I should be seeing 93 linked to this access token) with the following query as the `media` table is the only one being populated, yesterday there were 37 and today 63.  I have no idea if this is helpful but it is some information:

```
WITH 
usernames_by_date AS (
SELECT 
username, 
EXTRACT(DATE FROM _airbyte_emitted_at) AS ingest_date,
COUNT(*)
FROM `[project_id].[dataset_id].media`
GROUP BY username, ingest_date
),

distinct_usernames_by_date AS (
SELECT 
ingest_date,
COUNT (DISTINCT username) AS distinct_usernames
FROM usernames_by_date
GROUP BY ingest_date
ORDER BY ingest_date DESC
)


SELECT * FROM distinct_usernames_by_date
```

**Comment by yevhenii-ldv** [comment]
Hello, @jimbeepbeep.
Worked on this error, I was able to reproduce something like this only once.
I'm not sure yet, but there is a suspicion that the error occurs when the request limit is exceeded. We handle this case, postpone and continue to make requests for data, but Facebook for some reason sometimes returns not the standard error about "exceeding the rate limit", but the following error:
```  Status:  400    
  Response:
    {
      "error": {
        "message": "Unsupported get request. Object with ID '18066122755137560' does not exist, cannot be loaded due to missing permissions, or does not support this operation. Please read the Graph API documentation at https://developers.facebook.com/docs/graph-api",
        "type": "GraphMethodException",
        "code": 100,
        "error_subcode": 33,
        "fbtrace_id": "AXd8qVqHmYsMgTT3LS4c5fF"
      }
    }
```

In order to confirm my suspicions and start processing this error, I will ask you to use the Facebook developer page in order to test requests that failed and we received 400 errors from Facebook.
To do this, do the following:

1. Go to the [Facebook developer page](https://developers.facebook.com/tools/explorer/).
2. Indicate your active token in the corresponding cell (it is on the right). After you insert your active token, do **not** press the "Generate Access Token" button, as it will generate a new (not needed for verification) token.
3.  In the request field, specify several URLs that I will attach below (these are the links on which the script broke). And press the "Submit" button.


_**These are the links that must be specified in the URL:**_
```
17889247010206235/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
18102518305133212/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
17875703888498963/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
17886664972860586/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
17898104314850774/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
18066122755137560/?fields=thumbnail_url,ig_id,media_url,permalink,id,owner,shortcode,media_type,username,timestamp
```

If everything works out and the data is returned to you without errors, then my assumptions were confirmed and we will begin to implement additional error handling from Facebook.

I expect information from you. Thank you for contacting us.

**Comment by jimbeepbeep** [comment]
Thanks so much @yevhenii-ldv, I have tested these URLs in the API explorer with the current, live access token we use in Airbyte and can confirm that all of these return data without any errors.

I have also checked the App in the Facebook App Dashboard and we are apparently running at 7% capacity (93% available) and the average activity chart for the past 7 days looks like this:

![Screen Shot 2021-06-17 at 9 14 07 pm](https://user-images.githubusercontent.com/34912260/122460263-31b4e880-cfb2-11eb-95da-f4680bf683f7.png)
(Blue: Call Count Usage Rate, Green: Total Time Usage Rate, Purple: Total CPU Time Usage Rate).  Peaks are as expected (lots more calls when trying to test and fix).
